### ISSUE TITLE
Running Twice/Duplicate On Windows

### ISSUE THREAD

**Issue by c9845** [tdewolff/minify#527]
I have two minify commands, one each watching my script.js and styles.css file.  I am using `-w` for watching for changes to run the minification.  On Windows (10), the minification is run twice for each save of the non-minified file.  This occurs for both JS and CSS. 
 How do I know this? When using `-v` for verbose logging, the `INFO:` line is output twice; once with, usually, near 0 runtime and Inf MB/s speed, and the second with a realistic time and speed.

I have the same repo, go version, and minify version on macOS and the minification only runs once per file save, so this looks like a strictly Windows issue.

This does not impact performance or the actual minifying, it is just odd that minifying occurs twice.  It does not seem like just a logging bug either since sometimes the output does show a slight amount of time used by the "first run".

Example of logging output:
```
INFO: (       0s,    0 B, 100.0%, Inf MB/s) - website/static/js/script.js to website/static/js/script.min.js
INFO: (16.7102ms, 461 kB,  34.6%,  80 MB/s) - website/static/js/script.js to website/static/js/script.min.js
INFO: (    569µs,    0 B, 100.0%,    0 B/s) - website/static/js/script.js to website/static/js/script.min.js
INFO: (16.8865ms, 461 kB,  34.6%,  79 MB/s) - website/static/js/script.js to website/static/js/script.min.js
INFO: (       0s,    0 B, 100.0%, Inf MB/s) - website/static/js/script.js to website/static/js/script.min.js
INFO: (15.6304ms, 461 kB,  34.6%,  85 MB/s) - website/static/js/script.js to website/static/js/script.min.js
```

I tried to dig into the source code a bit but didn't have much luck (or time) in hunting down a cause.  I suspect there is something funky going on with how Windows reports file saves but I am not sure.

---

go version 1.19 (for both Windows and macOS).
Windows 10, macOS macOS Big Sur.
Latest minify (installed via go install github.com/tdewolff/minify/cmd/minify@latest).

Commands: 
- `minify -w -v -o website/static/js/script.min.js website/static/js/script.js`
- `minify -w -v -o website/static/css/styles.min.css website/static/css/styles.css`

**Comment by tdewolff** [comment]
Thanks for the issue. This is almost for sure a problem with fsnotify and how minify operates it. It is fairly hard to make it work on all operating systems the same, and it also depends on how they save the file (some programs overwrite the old file, some move the old file and write to a new one, some write to a temporary file, then move over the old file). I will look into it.

**Comment by c9845** [comment]
I think I figured this out.  It depends on the OS and the text editor being used.  The basic cause is how file saves are handled: modifying the file in place versus saving a new file and then copy/replacing the existing file.

There should be a way to "debounce" this using a timer per the suggestion here: https://github.com/fsnotify/fsnotify/issues/122#issuecomment-1065925569

**Comment by tdewolff** [comment]
Yes, great point. There should be a delay of 100ms per file from the last change before minifying to make sure that all intermediate write actions have completed. However, this has already been implemented! Every time a change happens (write, create, ...), minify waits 100ms before minifying. I have no clue as to why this is not sufficient in your case. Either it takes longer than 100ms (is the file very big?) or some other file renaming trick is happening...

**Comment by c9845** [comment]
The issue is tied into code editors on Windows only from what I can tell.  If I edit a non-minified file in VSCode or Sublime the "file changed" event fires twice and thus the minification runs twice.  If I edit the file in Notepad, the event and minification run once.

--- 

This gets a bit strange now... I forked your repo and `go build`-ed the binary.  When I run `minify` built from source, only a single event fires and minification occurs once.  I double checked I forked and built off the latest commit and I reinstalled via `go install github....` from your docs.  

I also see some extra logging when using the binary via `go install github...`.  Specifically, I see "INFO:" starting each line.  Clearly there is some slight difference in the binaries I am installing/building/using.

**Comment by tdewolff** [comment]
Those sound like some really old changes, when running `minify --version`, what does the one return with which you wrote this issue?

**Comment by c9845** [comment]
When minify if installed via the `go install github...` command, it shows "minify version master".

When minify is built from source (forked repo), it shows "built from source" (matches the code in `cmd/minify/main.go`, as expected).

**Comment by tdewolff** [comment]
It sounds like `go install` is installing a very old version! I'm wondering which version...anyways can you try and install `go install github.com/tdewolff/minify@latest`? Since it seems to work well when building from source, t seems to me that the bug is resolved in the latest version of minify.

**Comment by c9845** [comment]
I have tried `go install github...`.  This installs a version with `--version` printing "minify version master".  Also, I assume in you meant `go install github.com/tdewolff/minify/cmd/minify@latest` since `go install` does not work at the root of the repository.

I just tried it again and I get the same "minify version master" version.  I investigated this a bit and checked `$GOPATH` and it looks like v2.3.6 is being installed, not the latest v2.12.1.  When I go to https://pkg.go.dev/github.com/tdewolff/minify it is showing v2.3.6 so `go install` is grabbing what it is told to!

This led me to a solution: `go install github.com/tdewolff/v2/minify/cmd/minify@latest` (note the v2) installs a version that prints "minify built from source" and the downloaded source code shows v2.12.1.  So, the README in `cmd/minify` needs to be updated to add the `v2` to the path (pull request submitted).

Working with new v2, v3, etc. version of go modules sucks (a lot of the time), I know!
